R/bootstrap.R

```diff
@@ -1026,119 +1026,72 @@ buildNavItem <- function(divTag, tabsetId, index) {
   divTag$attribs$title <- NULL
   list(
     divTag = divTag,
-    liTag = tagFunction(function() {
-      navItem <- if ("3" %in% getCurrentVersion()) bs3NavItem else bs4NavItem
-      navItem(id, title, value, icon, active)
-    })
-  )
-}
-
-buildDropdown <- function(divTag, tabset) {
-  title <- divTag$title
-  value <- divTag$menuName
-  icon <- getIcon(iconClass = divTag$iconClass)
-  active <- containsSelectedTab(divTag$tabs)
-  list(
-    # list of tab content divs from the child tabset
-    divTag = tabset$content$children,
-    liTag = tagFunction(function() {
-      if ("3" %in% getCurrentVersion()) {
-        bs3NavItemDropdown(title, value, icon, active, tabset$navList)
-      } else {
-        # In BS4, dropdown nav anchors can't be wrapped in a <li> tag
-        # and also need .nav-link replaced with .dropdown-item to be
-        # styled sensibly
-        items <- tabset$navList
-        items$children <- lapply(items$children, function(x) {
-          # x should be a tagFunction() due to the else block below
-          x <- if (inherits(x, "shiny.tag.function")) x() else x
-          # Replace <li class="nav-item"><a class="nav-link"></a></li>
-          #    with <a class="dropdown-item"></a>
-          if (tagHasClass(x, "nav-item")) {
-            x <- x$children[[1]]
-            x$attribs$class <- "dropdown-item"
-          }
-          x
-        })
-        bs4NavItemDropdown(title, value, icon, active, items)
+    liTag = tagAddRenderHook(
+      liTag(id, title, value, icon),
+      function(x) {
+        if (isTRUE(getCurrentVersion() >= 4)) {
```

Comment:
Looking at `getCurrentVersion`, it looks like it can return a string in some cases. That function should be modified to always return a number.

Also, the name `getCurrentVersion` is confusing -- how about `getCurrentThemeVersion`?